packages/node/src/methods/state-channel/deposit/operation.ts

```diff
@@ -0,0 +1,148 @@
+import { StateChannel } from "@counterfactual/machine";
+import {
+  AssetType,
+  Node,
+  SolidityABIEncoderV2Struct
+} from "@counterfactual/types";
+import { AddressZero, One, Zero } from "ethers/constants";
+import { BigNumber } from "ethers/utils";
+
+import { RequestHandler } from "../../../request-handler";
+import { DepositConfirmationMessage, NODE_EVENTS } from "../../../types";
+import { getPeersAddressFromChannel } from "../../../utils";
+import { ERRORS } from "../../errors";
+
+export interface ETHBalanceRefundAppState extends SolidityABIEncoderV2Struct {
+  recipient: string;
+  multisig: string;
+  threshold: BigNumber;
+}
+
+export async function installBalanceRefundApp(
+  requestHandler: RequestHandler,
+  params: Node.DepositParams
+) {
+  const {
+    publicIdentifier,
+    instructionExecutor,
+    networkContext,
+    store
+  } = requestHandler;
+
+  const [peerAddress] = await getPeersAddressFromChannel(
+    publicIdentifier,
+    store,
+    params.multisigAddress
+  );
+  const initialState: ETHBalanceRefundAppState = {
+    recipient: publicIdentifier,
+    multisig: params.multisigAddress,
+    threshold: Zero
+  };
+  const stateChannel = await store.getStateChannel(params.multisigAddress);
+  const stateChannelsMap = await instructionExecutor.runInstallProtocol(
+    new Map<string, StateChannel>([
+      // TODO: (architectural decision) Should this use `getAllChannels` or
+      //       is this good enough? InstallProtocol only operates on a single
+      //       channel, anyway. PR #532 might make this question obsolete.
+      [stateChannel.multisigAddress, stateChannel]
+    ]),
+    {
+      initialState,
+      initiatingAddress: publicIdentifier,
+      respondingAddress: peerAddress,
+      multisigAddress: stateChannel.multisigAddress,
+      aliceBalanceDecrement: Zero,
+      bobBalanceDecrement: Zero,
+      signingKeys: stateChannel.getNextSigningKeys(),
+      terms: {
+        // TODO: generalize
+        assetType: AssetType.ETH,
+        // TODO: is this limit right?
+        limit: params.amount,
+        token: AddressZero
+      },
+      appInterface: {
+        addr: networkContext.ETHBalanceRefund,
+        stateEncoding:
+          "tuple(address recipient, address multisig,  uint256 threshold)",
+        actionEncoding: undefined
+      },
+      // TODO: what should the default timeout be here?
+      defaultTimeout: One.toNumber()
+    }
+  );
+
+  await store.saveStateChannel(stateChannelsMap.get(params.multisigAddress)!);
```

Comment:
avoid `!` when possible an do an error check

Reply:
This is mostly copy pasted from the already-existing install code.